doc/report/TUDreport.tex

```diff
@@ -0,0 +1,538 @@
+            \item Define custom filters to transform raw data inside views into an appropriate representation. For example, a custom filter could be applied to the value 4\,239\,283 in order to interpret it as a file size, e.g., 4.04 MB. Filters help to stick to the DRY programming principle and also keep logic and complex statements separate from views.
+            \item Configure third party services and modules. For example, the application uses the \texttt{angular-permission}\footnote{\url{https://github.com/Narzerus/angular-permission}} library to handle access control inside the application according to the users competence (see~\ref{sec:tech:implementation:acl:angular-permission}).
+        \end{itemize}
+    For further information on the concepts of AngularJS, please refer to their official documentation.\footnote{\url{https://docs.angularjs.org/guide/concepts}}
+
+  \subsection{Controllers}
+  \label{sec:tech:implementation:angularjs:controllers}
+
+    % TODO: Add a figure to describe how controllers interact with the view and includes services
+    In AngularJS a controller is usually associated with a view and is responsible to manage the logic behind it, i.e., it provides the data that is being displayed by the view and react to user interactions. The data exchange between the controller and the view happens implicitly through the built-in \texttt{\$scope} service\footnote{See section~\ref{sec:tech:implementation:angularjs:app} for a simple explaination of a service.} by using two-way data binding, i.e., whenever a value changes inside the view (through user interaction), the changes are reflected in the controller and vice versa. More information on controllers and the important concept of two-way data binding can be found in the official documentation.\footnote{\url{https://docs.angularjs.org/guide/controller} and \url{https://docs.angularjs.org/guide/databinding}}
+
+    Controllers are defined inside the \emph{client/js/controllers/} directory.
+
+  %\subsection{Layout and Templates}
+  %\label{sec:tech:implementation:angularjs:layout}
+
+    % TODO: Make a figure that describes the layout template and its embedded parts
+
+  \section{Role Based Access Control (RBAC)}
+  \label{sec:tech:implementation:rbac}
+
+    Access control has to happen in both parts of the application, server and client. On the server side, the application needs to ensure that data can only be accessed and manipulated by users that are actually authorized to do so, i.e., a user should not be able to edit the submission of someone else, unless he is chair of the corresponding conference. On the client side, permissions also need to be checked, e.g., to dynamically display administration links in the navigation bar based on whether the user is chair of the conference.
```

Comment:
... of the application, namely server and client. (vielleicht namely reinschreiben?)